File: app/src/main/java/com/github/anrimian/musicplayer/ui/common/error/parser/ErrorParser.java
package com.github.anrimian.musicplayer.ui.common.error.parser;

import android.content.Context;

import androidx.annotation.StringRes;

import com.github.anrimian.musicplayer.ui.common.error.ErrorCommand;

/**
 * Created on 29.10.2017.
 */

public abstract class ErrorParser {

    private final Context context;

    public ErrorParser(Context context) {
        this.context = context;
    }

    protected String getString(@StringRes int resId) {
        return context.getString(resId);
    }

    protected String getString(@StringRes int resId, Object... formatArgs) {
        return context.getString(resId, formatArgs);
    }

    protected ErrorCommand error(@StringRes int resId) {
        return new ErrorCommand(getString(resId));
    }

    public abstract ErrorCommand parseError(Throwable throwable);

    public abstract void logError(Throwable throwable);
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/common/error/parser/ErrorParser.kt
package com.github.anrimian.musicplayer.ui.common.error.parser

import android.content.Context
import androidx.annotation.StringRes
import com.github.anrimian.musicplayer.ui.common.error.ErrorCommand

/**
 * Created on 29.10.2017.
 */
abstract class ErrorParser(private val context: Context) {

    protected fun getString(@StringRes resId: Int): String {
        return context.getString(resId)
    }

    protected fun getString(@StringRes resId: Int, vararg formatArgs: Any): String {
        return context.getString(resId, *formatArgs)
    }

    protected fun error(@StringRes resId: Int): ErrorCommand {
        return ErrorCommand(getString(resId))
    }

    abstract fun parseError(throwable: Throwable): ErrorCommand

    abstract fun logError(throwable: Throwable)

}